dvc/remote/ssh/__init__.py

```diff
@@ -38,27 +32,36 @@ class RemoteSSH(RemoteBASE):
 
     def __init__(self, repo, config):
         super(RemoteSSH, self).__init__(repo, config)
-        self.url = config.get(Config.SECTION_REMOTE_URL, "ssh://")
-
-        parsed = urlparse(self.url)
-        self.host = parsed.hostname
 
-        user_ssh_config = self._load_user_ssh_config(self.host)
+        url = config.get(Config.SECTION_REMOTE_URL)
+        if url:
+            parsed = urlparse(url)
+            user_ssh_config = self._load_user_ssh_config(parsed.hostname)
+
+            host = user_ssh_config.get("hostname", parsed.hostname)
+            user = (
+                config.get(Config.SECTION_REMOTE_USER)
+                or parsed.username
+                or user_ssh_config.get("user")
+                or getpass.getuser()
+            )
+            port = (
+                config.get(Config.SECTION_REMOTE_PORT)
+                or parsed.port
+                or self._try_get_ssh_config_port(user_ssh_config)
+                or self.DEFAULT_PORT
+            )
+            self.path_info = self.path_cls.from_parts(
+                scheme=self.scheme,
+                host=host,
+                user=user,
+                port=port,
+                path=parsed.path,
+            )
+        else:
+            self.path_info = None
```

Comment:
is it actually an exception?

Reply:
The thing is we need to instantiate remote even when it doesn't have path, because we do everything with outputs in remote methods. It was like this before, just wasn't as obvious - we created dummy path objects, which were never used, now that is at least enforced.